core/src/main/scala/cats/evidence/As.scala

```diff
@@ -0,0 +1,268 @@
+package cats
+package evidence
+
+import arrow.Category
+
+/**
+ * As substitutability: A better `<:<`
+ * 
+ * This class exists to aid in the capture proof of subtyping
+ * relationships, which can be applied in other context to widen other
+ * type
+ *
+ * `A As B` holds whenever `A` could be used in any negative context
+ * that expects a `B`.  (e.g. if you could pass an `A` into any
+ * function that expects a `B` as input.)
+ * 
+ * This code was ported directly from scalaz to cats using this version from scalaz:
+ * https://github.com/scalaz/scalaz/blob/a89b6d63/core/src/main/scala/scalaz/As.scala
+ * 
+ *  The original contribution to scalaz came from Jason Zaugg
+ */
+sealed abstract class As[-A, +B] {
+  def apply(a: A): B = As.witness(this)(a)
+
+  /**
+   * Use this subtyping relationship to replace B with a value of type
+   * A in a contravariant context.  This would commonly be the input
+   * to a function, such as F in: `type F[-B] = B => String`. In this
+   * case, we could use A As B to turn an F[B] Into F[A].
+   */
+  def subst[F[-_]](p: F[B]): F[A]
+
+  final def andThen[C](that: As[B, C]): As[A, C] = As.trans(that, this)
+
+  final def compose[C](that: As[C, A]): As[C, B] = As.trans(this, that)
+
+  /**
+   * Use this relationship to widen the output type of a Function1
+   */
+  def onF[X](fa: X => A): X => B = As.co2_2[Function1, B, X, A](this)(fa)
+}
+
+sealed abstract class AsInstances {
+  import As._
+
+  /*
+   * Subtyping forms a category 
+   */
+  implicit val liskov: Category[<~<] = new Category[<~<] {
+    def id[A]: (A <~< A) = refl[A]
+
+    def compose[A, B, C](bc: B <~< C, ab: A <~< B): (A <~< C) = trans(bc, ab)
+  }
+}
+
+object As extends AsInstances {
+  /**
+   * Lift Scala's subtyping relationship 
+   */
+  implicit def isa[A, B >: A]: A <~< B =
+    new (A <~< B) {
+      def subst[F[-_]](p: F[B]): F[A] = p
+    }
+
+  /**
+   * We can witness the relationship by using it to make a substitution *
+   */
+  implicit def witness[A, B](lt: A <~< B): A => B = 
+    lt.subst[-? => B](identity)
+
+  /**
+   * Subtyping is reflexive 
+   */
+  implicit def refl[A]: (A <~< A) =
+    new (A <~< A) {
+      def subst[F[-_]](p: F[A]): F[A] = p
+    }
+
```

Comment:
We don't have a method to materialize a `<:<` as needed. I think we should. Also a similar unsafe method to go from `<:<` to As.